docs/source/process.rst

```diff
@@ -0,0 +1,570 @@
+    'label': ClassLabel(num_classes=2, names=['not_equivalent', 'equivalent'], names_file=None, id=None),
+    'idx': Value(dtype='int32', id=None)}
+    >>> from datasets import ClassLabel, Value
+    >>> new_features = dataset.features.copy()
+    >>> new_features["label"] = ClassLabel(names=['negative', 'positive'])
+    >>> new_features["idx"] = Value('int64')
+    >>> dataset = dataset.cast(new_features)
+    >>> dataset.features
+    {'sentence1': Value(dtype='string', id=None),
+    'sentence2': Value(dtype='string', id=None),
+    'label': ClassLabel(num_classes=2, names=['negative', 'positive'], names_file=None, id=None),
+    'idx': Value(dtype='int64', id=None)}
+
+.. tip::
+
+    Casting only works if the original feature type and new feature type are compatible. For example, you can cast a column with the feature type ``Value('int32')`` to ``Value('bool')`` if the original column only contains ones and zeros. 
+
+``Flatten``
+^^^^^^^^^^^
+
+Sometimes a column can be a nested structure of several types. In these scenarios, use :func:`datasets.Dataset.flatten` to extract the subfields into their own separate columns. Take a look at the nested structure below from the SQuAD dataset:
+
+    >>> from datasets import load_dataset
+    >>> dataset = load_dataset('squad', split='train')
+    >>> dataset.features
+    {'answers': Sequence(feature={'text': Value(dtype='string', id=None), 'answer_start': Value(dtype='int32', id=None)}, length=-1, id=None),
+    'context': Value(dtype='string', id=None),
+    'id': Value(dtype='string', id=None),
+    'question': Value(dtype='string', id=None),
+    'title': Value(dtype='string', id=None)}
+
+The **answers** field contains two subfields: **text** and **answer_start**. After you use :func:`datasets.Dataset.flatten`, these subfields are now their own independent columns:
+
+    >>> flat_dataset = dataset.flatten()
+    >>> flat_dataset
+    Dataset({
+        features: ['id', 'title', 'context', 'question', 'answers.text', 'answers.answer_start'],
+        num_rows: 87599
+    })
+
+Concatenate
+------------
+
+Separate datasets can be concatenated if they share the same column types. Concatenate datasets with :func:`datasets.concatenate_datasets`:
+
+   >>> from datasets import concatenate_datasets, load_dataset
+   >>>
+   >>> bookcorpus = load_dataset("bookcorpus", split="train")
+   >>> wiki = load_dataset("wikipedia", "20200501.en", split="train")
+   >>> wiki = wiki.remove_columns("title")  # only keep the text
+   >>>
+   >>> assert bookcorpus.features.type == wiki.features.type
+   >>> bert_dataset = concatenate_datasets([bookcorpus, wiki])
+
+.. seealso::
+
+    You can also mix several datasets together by taking alternating examples from each one to create a new dataset. This is known as interleaving datasets, and you can use it with :func:`datasets.interleave_datasets`.  See the ``IterableDataset`` section for an example of how it's used.
```

Comment:
Ultimately both `interleave_datasets` and `concatenate_datasets` will work on regular `Dataset` and streaming `IterableDataset` objects

Reply:
Added sentence to specify that both methods will work with a regular `Dataset` and streaming `IterableDataset` object :)